libsolidity/ast/Types.cpp

```diff
@@ -3014,13 +3016,17 @@ MemberList::MemberMap MagicType::nativeMembers(ContractDefinition const*) const
 			{"gaslimit", make_shared<IntegerType>(256)}
 		});
 	case Kind::Message:
-		return MemberList::MemberMap({
+	{
+		std::vector<MemberList::Member> members = {
 			{"sender", make_shared<IntegerType>(160, IntegerType::Modifier::Address)},
-			{"gas", make_shared<IntegerType>(256)},
 			{"value", make_shared<IntegerType>(256)},
 			{"data", make_shared<ArrayType>(DataLocation::CallData)},
 			{"sig", make_shared<FixedBytesType>(4)}
-		});
+		};
+		if (!v050)
+			members.emplace_back("gas", make_shared<IntegerType>(256));
+		return MemberList::MemberMap(members);
```

Comment:
The type of `members` should be identical to `memberList::MemberMap` shouldn't it?